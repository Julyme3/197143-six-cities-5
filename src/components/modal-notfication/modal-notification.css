.modal {
  position: fixed;
  top: 50px;
  right: 50%;
  transform: translateX(50%);
  width: 500px;
  margin: 50px auto;
  border: 1px solid rgba(0,0,0,.2);
  border-radius: .3rem;
  background-color: #fff;
  box-shadow: 0 3px 7px rgba(0, 0, 0, 0.3);
}

.modal-header {
  padding: 1rem;
  border-bottom: 1px solid #e9ecef;
}

.modal-title {
  font-size: 1.25rem;
  margin: 0;
  color: red;
}

.modal-body {
  padding: 1rem;
}

.modal.isHidden {
  display: none;
}
