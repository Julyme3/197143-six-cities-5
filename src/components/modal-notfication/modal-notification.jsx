import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import './modal-notification.css';
import {getContent, getIsError} from "../../store/reducers/notification/selectors";
import {connect} from "react-redux";

class ModalNotification extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {showError, content} = this.props;
    const modalClassHidden = showError ? `` : `isHidden`;

    return (
      <div className={`modal ${modalClassHidden}`} tabIndex="-1">
        <div className="modal-dialog" role="document">
          <div className="modal-content">
            <div className="modal-header">
              <h5 className="modal-title" id="exampleModalLabel">Error</h5>
            </div>
            <div className="modal-body">
              Error message: {content}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

ModalNotification.propTypes = {
  showError: PropTypes.bool,
  content: PropTypes.string,
};

const mapStateToProps = (state) => ({
  showError: getIsError(state),
  content: getContent(state),
});

export {ModalNotification};

export default connect(mapStateToProps)(ModalNotification);
