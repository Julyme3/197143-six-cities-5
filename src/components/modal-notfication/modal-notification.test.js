import React from "react";
import renderer from "react-test-renderer";
import {ModalNotification} from "./modal-notification";

const noop = () => {};
const content = `Response status: 401 onUnauthorized`;

test(`ModalNotification render`, () => {
  const tree = renderer
    .create(
        <ModalNotification
          setNotification={noop}
          showError={true}
          content={content}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
