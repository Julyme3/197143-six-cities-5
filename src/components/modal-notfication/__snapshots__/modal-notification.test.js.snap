// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalNotification render 1`] = `
<div
  className="modal "
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="exampleModalLabel"
        >
          Error
        </h5>
      </div>
      <div
        className="modal-body"
      >
        Error message: 
        Response status: 401 onUnauthorized
      </div>
    </div>
  </div>
</div>
`;
